Signal/src/ViewControllers/OWSTableViewController.m

```diff
@@ -0,0 +1,226 @@
+    OWSAssert(title.length > 0);
+    
+    OWSTableItem *item = [OWSTableItem new];
+    item.itemType = OWSTableItemTypeAction;
+    item.actionBlock = actionBlock;
+    item.title = title;
+    return item;
+}
+
+@end
+
+#pragma mark -
+
+NSString * const kOWSTableCellIdentifier = @"kOWSTableCellIdentifier";
+
+@implementation OWSTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self.navigationController.navigationBar setTranslucent:NO];
+}
+
+- (instancetype)init
+{
+    return [super initWithStyle:UITableViewStyleGrouped];
+}
+
+- (void)loadView
+{
+    [super loadView];
+    
+    OWSAssert(self.contents);
+
+    self.title = self.contents.title;
+    
+    OWSAssert(self.tableView);
+    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:kOWSTableCellIdentifier];
+}
+
+- (OWSTableSection *)sectionForIndex:(NSInteger)sectionIndex
+{
+    OWSAssert(self.contents);
+    OWSAssert(sectionIndex >= 0 && sectionIndex < (NSInteger) self.contents.sections.count);
+    
+    OWSTableSection *section = self.contents.sections[(NSUInteger) sectionIndex];
+    return section;
+}
+
+- (OWSTableItem *)itemForIndexPath:(NSIndexPath *)indexPath
+{
+    OWSAssert(self.contents);
+    OWSAssert(indexPath.section >= 0 && indexPath.section < (NSInteger) self.contents.sections.count);
+    
+    OWSTableSection *section = self.contents.sections[(NSUInteger) indexPath.section];
+    OWSAssert(indexPath.item >= 0 && indexPath.item < (NSInteger) section.items.count);
+    OWSTableItem *item = section.items[(NSUInteger) indexPath.item];
+    
+    return item;
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    OWSAssert(self.contents);
+
+    OWSAssert(self.contents.sections.count > 0);
+    return (NSInteger) self.contents.sections.count;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)sectionIndex {
+    OWSTableSection *section = [self sectionForIndex:sectionIndex];
+    OWSAssert(section.items.count > 0);
+    return (NSInteger) section.items.count;
+}
+
+- (nullable NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)sectionIndex
+{
+    OWSTableSection *section = [self sectionForIndex:sectionIndex];
+    return section.title;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    OWSTableItem *item = [self itemForIndexPath:indexPath];
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kOWSTableCellIdentifier];
+    OWSAssert(cell);
+    
+    cell.textLabel.text = item.title;
+    
+    return cell;
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+
+    OWSTableItem *item = [self itemForIndexPath:indexPath];
+    if (item.itemType == OWSTableItemTypeAction) {
```

Comment:
* `itemForIndexPath` should never return nil.
* I've changed this condition in a subsequent PR.